testdata/sqllogictests/drop.slt

```diff
@@ -292,3 +292,26 @@ drop schema glare_catalog;
 
 statement error
 drop table glare_catalog.databases;
+
+# Delete table data only when planning succeeds for all tables.
+
+statement ok
+create table drop_data_check as values (1);
+
+query I
+select * from drop_data_check
+----
+1
+
+statement error
+drop table drop_data_check, table_does_not_exist;
+
+# Since the above query fails due to a table not existing, we should _not_ drop
+# any data.
+#
+# Note that this doesn't guarantee that we drop data in the correct order since
+# we don't have a guarantee on when the background job runs.
```

Comment:
We can't really guarantee that the data isn't dropped here either. Maybe the background job hasn't still run 🤔

Does this test error on main?

Reply:
Yeah this was the easiest test case to write, but definitely not comprehensive.

I was able to get this to fail on main. It was easier to repeat when running locally:

```
GlareDB (v0.2.1)
Type \help for help.
Using in-memory catalog
〉\open db
Created new session. New database path: db
〉create table drop_data_check as values (1);
create table
〉drop table drop_data_check, table_does_not_exist;
Error: Failed Metastore request: Missing database object; schema: public, name: table_does_not_exist
〉select * from drop_data_check;
Error: Error during planning: Unable to fetch table provider for 'drop_data_check': Failed to create table provider for 'drop_data_check': Not a Delta table: No snapshot or version 0 found, perhaps /Users/sean/Code/github.com/glaredb/glaredb/db/databases/00000000-0000-0000-0000-000000000000/tables/20000/ is an empty dir?
```